src/Microsoft.Performance.Toolkit.Engine/EngineCreateInfo.cs

```diff
@@ -38,6 +44,50 @@ public EngineCreateInfo(ReadOnlyDataSourceSet dataSources)
             Guard.NotNull(dataSources, nameof(dataSources));
 
             this.DataSources = dataSources;
+            this.optionsResolver = GlobalProcessingOptionsResolver.Default;
+        }
+
+        /// <summary>
+        ///     <see cref="ProcessorOptions"/> to use for all processing sources and data source groups during processing.
+        /// </summary>
+        /// <param name="globalProcessorOptions"> <see cref="ProcessorOptions"/> to use for all processing.</param>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="globalProcessorOptions"/> is <c>null</c>.
+        /// </exception>
+        public void WithProcessorOptions(ProcessorOptions globalProcessorOptions)
+        {
+            Guard.NotNull(globalProcessorOptions, nameof(globalProcessorOptions));
+
+            this.WithProcessingOptions(new GlobalProcessingOptionsResolver(globalProcessorOptions));
+        }
+
+        /// <summary>
+        ///     Set <see cref="ProcessorOptions"/> per <see cref="IProcessingSource"/>.
+        /// </summary>
+        /// <param name="processingSourceOptionsMap">A map of <see cref="IProcessingSource"/>s with <see cref="ProcessorOptions"/> to pass it.</param>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="processingSourceOptionsMap"/> is <c>null</c>.
+        /// </exception>
+        public void WithProcessorOptions(IDictionary<IProcessingSource, ProcessorOptions> processingSourceOptionsMap)
```

Comment:
How does someone calling this method get the `IProcessingSource` instance(s) that would be in this dictionary?